packages/manager/modules/log-to-customer/src/live-tail/controller.js

```diff
@@ -25,20 +31,100 @@ export default class LogToCustomerCtrl {
       createAccount: `${this.dbaasLogsURL}/order`,
     };
 
+    this.kind = this.kindInitValue;
     this.hasAtLeastOneDataStream = false;
+    this.errorMessage = null;
 
-    this.kind = this.kindInitValue;
+    this.init();
+  }
 
-    this.LogToCustomer.getDataStreams()
-      .then(({ streams, streamCount }) => {
-        this.logAccountStreams = streams;
-        this.hasAtLeastOneDataStream = streamCount > 0;
+  init() {
+    this.resetErrorMessage();
+    this.loading = true;
+    this.streamSubscriptions = [];
+
+    this.$q
+      .all([this.getDataStreams(), this.getSubscribedStreams()])
+      .catch(({ data }) => {
+        this.errorMessage = data.message;
       })
       .finally(() => {
         this.loading = false;
       });
   }
 
+  resetErrorMessage() {
+    this.errorMessage = null;
+  }
+
+  deleteLogSubscription(subscription) {
+    this.resetErrorMessage();
+    // eslint-disable-next-line no-param-reassign
+    subscription.isLoading = true;
+
+    this.LogToCustomerService.delete(
+      `${this.logSubscriptionUrl}/${subscription.subscriptionId}`,
+    )
+      .then(({ data }) => {
+        this.LogToCustomerService.pollOperation(
+          subscription.serviceName,
+          data,
+        ).then(() => this.init());
+      })
+      .catch(({ data }) => {
+        this.errorMessage = data.message;
+      })
+      .finally(() => {
+        // eslint-disable-next-line no-param-reassign
+        subscription.isLoading = false;
+      });
+  }
+
+  getDataStreams() {
+    return this.LogToCustomerService.getDataStreams().then(
+      ({ streams, streamCount }) => {
+        this.logAccountStreams = streams;
+        this.hasAtLeastOneDataStream = streamCount > 0;
+      },
+    );
+  }
+
+  async getSubscribedStreams() {
+    try {
+      const data = await this.LogToCustomerService.icebergQuery(
+        this.logSubscriptionUrl,
+        { kind: this.kind },
+      );
+
+      const streamPromises = data.map(async (subscription) => {
+        const account = await this.LogToCustomerService.getLogAccount(
+          subscription.serviceName,
+        );
+        const stream = await this.LogToCustomerService.getSream(
+          subscription.serviceName,
+          subscription.streamId,
+        );
+        const url = await this.LogToCustomerService.getStreamUrl(
+          subscription.serviceName,
+          subscription.streamId,
+        );
```

Comment:
With `await` callback, all your calls are done synchronously whereas it's not necessary and therefore could be otpimized